### PR TITLE
Update to NullAway 0.12.6

### PR DESCRIPTION
**Pull request by msridhar**
Fix or suppress newly-reported issues

### CODE REVIEW

**Comment by ben-manes** [comment]
sgtm, thank you!

File: caffeine/src/test/java/com/github/benmanes/caffeine/cache/AsyncAsMapTest.java [hunk 1]
```diff
@@ -2594,7 +2595,7 @@ public void entrySet_remove_null(AsyncCache<Int, Int> cache, CacheContext contex
   @Test(dataProvider = "caches")
   public void entrySet_remove_nullKey(AsyncCache<Int, Int> cache, CacheContext context) {
     var future = Iterables.getFirst(cache.asMap().values(), context.absentValue().toFuture());
-    assertThat(cache.asMap().entrySet().remove(Maps.immutableEntry(null, future))).isFalse();
+    assertThat(cache.asMap().entrySet().remove(Maps.<@Nullable Int, CompletableFuture<Int>>immutableEntry(null, future))).isFalse();
```
Comment (msridhar) [inline caffeine/src/test/java/com/github/benmanes/caffeine/cache/AsyncAsMapTest.java:2598]: These explicit type arguments are needed due to limitation in NullAway's inference.  They kind of clutter the code, so we could just suppress instead

File: caffeine/src/test/java/com/github/benmanes/caffeine/cache/AsMapTest.java [hunk 2]
```diff
@@ -2792,7 +2792,7 @@ public void entrySet_remove_null(Map<Int, Int> map, CacheContext context) {
 
   @CacheSpec
   @CheckNoStats
-  @SuppressWarnings("MapEntry")
+  @SuppressWarnings({"MapEntry","NullAway"})
```
Comment (msridhar) [inline caffeine/src/test/java/com/github/benmanes/caffeine/cache/AsMapTest.java:2795]: Here I suppressed rather than adding explicit type arguments (compare with `AsyncAsMapTest` fixes).

**Comment by msridhar** [comment]
@ben-manes this passes NullAway 0.12.6 now, except for a couple of `Map.get` calls where I wasn't sure if they could actually return `null` so I didn't suppress; was hoping you could look at those.  All other suppressions are due to remaining limitations in NullAway's inference of type arguments for generic method calls.

Generally speaking probably good if you look over all the changes since I don't know this code well 🙂

File: caffeine/src/test/java/com/github/benmanes/caffeine/cache/AsyncAsMapTest.java (hunk 1 above)
Comment (ben-manes) [reply caffeine/src/test/java/com/github/benmanes/caffeine/cache/AsyncAsMapTest.java:2598]: either is fine with me

Comment (msridhar) [reply caffeine/src/test/java/com/github/benmanes/caffeine/cache/AsyncAsMapTest.java:2598]: Ok, let me change to suppress.  When we improve NullAway's inference we then should be able to detect the unnecessary suppression and remove it

**Comment by ben-manes** [comment]
thanks! I'll fix them. I believe in these cases I just add `requireNonNull` since it is implicit test state, so invalid if null (and that caught one or two mistakes!), and satisfies the checker.
